core/src/main/java/feign/template/Template.java

```diff
@@ -32,6 +32,13 @@
  */
 public class Template {
 
+  /*
+   * special delimiter for collection based expansion, in an attempt to avoid accidental splitting
+   * for resolved values. semi-colon was chosen because it is a reserved character that must be
+   * pct-encoded and should not appear unencoded.
+   */
+  static final String COLLECTION_DELIMITER = ";";
```

Comment:
Should we allow this to be configurable? I don't have a opinion on this.

Reply:
This is for our internal use.  User's can configure the output through the `CollectionFormat` enum already.  I chose semi-colon as to not interfere with those settings.  That turned out to be the cause of the bug, since comma was already taken.